mautic/mautic
# Bugfix for DateTime filters in reports

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | N
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

#### Description:
There is an issue with DateTime filters in reports, because in the backend the filters are applied using like, so the generated query is like this one: 

```
SELECT 
    l.id AS id,
    l.email AS email,
    l.firstname AS firstname,
    l.lastname AS lastname
FROM
    leads l
WHERE
    (l.date_added BETWEEN '2017-07-13 00:00:00' AND '2017-08-12 01:58:27')
        AND (l.date_identified LIKE '2017-07-13 19:50:07')
LIMIT 30 OFFSET 0
```
This PR fixes this by fetching the report filters and replacing LIKE with the correct operator.

#### Steps to reproduce the bug:
1. Create a contact report with a dynamic filter of "date identified >= 1/1/2017".  
2. Close and save report
3. Enter a new date for the filter
4. No results will be displayed
5. Check the logs and you will see the query is using LIKE to search for the date

#### Steps to test this PR:
1. Apply this PR
2. Repeat the steps to reproduce the bug
3. This time the report will show data (if there is any)
4. If you check the logs you will see that the query is using the correct operator.
